trunk/src/app/srs_app_recv_thread.cpp

```diff
@@ -596,6 +597,7 @@ srs_error_t SrsHttpRecvThread::cycle()
         SrsAutoFree(ISrsHttpMessage, req);
         
         if ((err = conn->pop_message(&req)) != srs_success) {
+	    trd_err = err;
```

Comment:
The err here is shared by trd_err and wrapped err, we should free the trd_err then copy err to it.